src/main/scala/zio/nio/ByteBuffer.scala

```diff
@@ -1,15 +1,18 @@
 package zio.nio
 
-import zio.{ Chunk, IO, ZIO }
+import zio.{ Chunk, IO, UIO, ZIO }
 import java.nio.{ BufferUnderflowException, ByteOrder, ReadOnlyBufferException, ByteBuffer => JByteBuffer }
 
-final class ByteBuffer(byteBuffer: JByteBuffer) extends Buffer[Byte](byteBuffer) {
+final class ByteBuffer(private val byteBuffer: JByteBuffer) extends Buffer[Byte](byteBuffer) {
```

Comment:
could you keep this public because we are using ByteBuffer in https://github.com/zio/zio-keeper/pull/44 or we should provide some way to create ByteBuffer from array and with given size.

```
object ByteBuffer {
   def allocateDirect(size: Int) = new zio.nio.ByteBuffer(java.nio.ByteBuffer.allocateDirect(size))
   def fromChunk(chunk: Chunk[Byte]) = new zio.nio.ByteBuffer(java.nio.ByteBuffer.wrap(chunk.toArray))
}
```

@quelgar what is your opinion?

Reply:
@pshemass The constructor is kept public. Only the field was made private, which was never publicly accessible in the first place. For your alternative request, those already exist within the `Buffer` companion object.

https://github.com/zio/zio-nio/blob/master/src/main/scala/zio/nio/Buffer.scala#L107

Reply:
OMG thanks I missed that completely!